### ISSUE TITLE
Instance generators fail for combinatorial auctions, independent set, facility location  

### ISSUE THREAD

**Issue by Sandbergo** [ds4dm/ecole#141]
## Describe the bug
Combinatorial auctions generator produces problems taking an unreasonable number of nodes to solve (about 20 000), independent set, facility location does not find the solution within reasonable time. Problem parameters are the same as the small problems from Gasse et al. (2019). I have no issue with the set cover problems. 

## Setting
- OS: Ubuntu 20.04.2 LTS x86_64
- Compiler: gcc 9.3.0
- Python version: 3.8.5
- Ecole version: 0.5.0

## To Reproduce
```python
import ecole

if __name__ == "__main__":
    scip_parameters = {'separating/maxrounds': 0, 'presolving/maxrestarts': 0, 'limits/time': 3600}

    default_env = ecole.environment.Branching(observation_function=ecole.observation.Pseudocosts(),
                                                information_function={"nb_nodes": ecole.reward.NNodes().cumsum(), 
                                                                      "time": ecole.reward.SolvingTime().cumsum()}, 
                                                scip_params=scip_parameters)

    generators = {
        'setcover':ecole.instance.SetCoverGenerator(n_rows=500, n_cols=1000, density=0.05),
        'cauctions': ecole.instance.CombinatorialAuctionGenerator(n_items=100, n_bids=500),
        'indset': ecole.instance.IndependentSetGenerator(n_nodes=500, graph_type="erdos_renyi", affinity=4),
        'facilities': ecole.instance.CapacitatedFacilityLocationGenerator(n_customers=100, n_facilities=100)
        }
    
    for problem_type, generator in generators.items():
        print(f'    Problem: {problem_type}')
        for instance_count, instance in zip(range(3), generator):
        
            observation, action_set, _, done, info = default_env.reset(instance)
            
            while not done:
                scores = observation
                action = action_set[observation[action_set].argmax()]
                observation, action_set, _, done, info = default_env.step(action)

            print(f"SCIP nb nodes  {int(info['nb_nodes']): >4d}  | SCIP time {info['time']: >6.2f} ")
```

## Expected behavior
Solving all instances within a few seconds.

## Additional context
Ecole built from source. SCIP 7.0.2.

**Comment by AntoinePrv** [comment]
@gasse @dchetelat Are we expecting to solve instances within a few seconds with `argmax` on `ecole.observation.Pseudocost`?
Is this the default SCIP branching strategy?

**Comment by Sandbergo** [comment]
I get similar performance from running the Configuring-environment and solving with `default_env.step({})` as in `examples/branching_imitation.ipynb`, however I saw there was an open issue with this method.

**Comment by AntoinePrv** [comment]
@Sandbergo thank you for raising this by the way. I'm not the best to answer and Maxime and Didier are busy this week, so it may take a bit.

**Comment by dchetelat** [comment]
Hi @Sandbergo, you're right, I just checked and the instances look very hard. 

```python
import ecole


scip_parameters = {'separating/maxrounds': 0, 'presolving/maxrestarts': 0, 'limits/time': 3600}
default_env = ecole.environment.Configuring(observation_function=None,
                                            information_function={"nb_nodes": ecole.reward.NNodes(), 
                                                                  "time": ecole.reward.SolvingTime()}, 
                                            scip_params=scip_parameters)

generator = ecole.instance.CombinatorialAuctionGenerator(n_items=100, n_bids=500)
instance = next(generator)

_, _, _, _, _ = default_env.reset(instance)
_, _, _, done, info = default_env.step({})
```
with
```
> info
{'nb_nodes': 10315.0, 'time': 43.645617139}
```
Maybe there's a bug in the generators, or something with SCIP hyperparameters. We'll look into it. Thanks for catching that!

**Comment by dchetelat** [comment]
Oh wait I think I already found out what the problem was! For `cauctions` is an additional generator hyperparameter, `add_item_prob`, which is set to 0.7 in the learn2branch paper code but by default at 0.9 in Ecole. If you set it to 0.7 you get reasonable instances.

```python
import ecole


scip_parameters = {'separating/maxrounds': 0, 'presolving/maxrestarts': 0, 'limits/time': 3600}
default_env = ecole.environment.Configuring(observation_function=None,
                                            information_function={"nb_nodes": ecole.reward.NNodes(), 
                                                                  "time": ecole.reward.SolvingTime()}, 
                                            scip_params=scip_parameters)

generator = ecole.instance.CombinatorialAuctionGenerator(n_items=100, n_bids=500, add_item_prob=0.7)
instance = next(generator)

_, _, _, _, _ = default_env.reset(instance)
_, _, _, done, info = default_env.step({})
```
and
```
> info
{'nb_nodes': 7.0, 'time': 2.369696843}
```

For `indset`, the problem seems to be that there's a discrepancy between what is reported in the paper and what we actually do (based on the code): namely the code says we generated Barabasi-Albert graphs but the paper says we used Erdös-Rényi graphs, which is an issue we'll have to discuss internally. If you switch the graphs to Barabasi-Albert the instances take reasonable time.

For `facilities` I didn't have any issues, the default generator seems to work fine.

**Comment by Sandbergo** [comment]
Thank you very much for your reply! 

With your help, `cauctions` and `indset` now seem to work for me (although with some problem instances being solved in ​around 500 nodes with the default brancher, which seems suspicious).

I unfortunately still have problems with `facilities`. I will keep you informed if I find anything.

**Comment by dchetelat** [comment]
Hi @Sandbergo, I was wrong, it turns out we have the same problem with `facilities` indeed. We think the problem with that one is not just an issue of hyperparameters, that we must have broken something while rewriting the code in C++ to make it faster. We'll investigate and keep you posted!

**Comment by Sandbergo** [comment]
Hello!

Yes, that seems probable based on my experiences as well, good luck with the debugging!

**Comment by AntoinePrv** [comment]
## CFLP Bug 1
https://github.com/ds4dm/ecole/blob/6bcb3926f9fe1d9334b6849a38713adbbd187003/libecole/src/instance/capacitated-facility-location.cpp#L230

@gasse said
> There seems to be a discrepancy from the original Python code here. The conversion of fixed_costs to an int seems to have been lost in the process. Is this intentional ?
https://github.com/ds4dm/learn2branch/blob/6af745f25c25007535e7fb5a1359b518528914e6/01_generate_instances.py#L475

@AntoinePrv said
So we have randint then a sqrt makes that into a float, and then we cast to int again?
I guess we missed that last part.

## CFLP Bug 2
https://github.com/ds4dm/ecole/blob/6bcb3926f9fe1d9334b6849a38713adbbd187003/libecole/src/instance/capacitated-facility-location.cpp#L193

@gasse said:
> The main tightening constraint, regarding total capacity >= total demand, is missing:
https://github.com/ds4dm/learn2branch/blob/master/01_generate_instances.py#L503
What does .astype(int) do ? floor ? ceil ? truncation ?

@AntoinePrv said:
> You're right

**Comment by AntoinePrv** [comment]
Closing this should be good in master now, and released in `v0.6.0` (upcoming).
